framework/core/js/src/common/components/Dropdown.tsx

```diff
@@ -78,15 +86,25 @@ export default class Dropdown extends Component {
       const $menu = this.$('.Dropdown-menu');
       const isRight = $menu.hasClass('Dropdown-menu--right');
 
+      const top = $menu.offset()?.top || 0;
+      const height = $menu.height() || 0;
+      const windowSrollTop = $(window).scrollTop() || 0;
+      const windowHeight = $(window).height() || 0;
+
       $menu.removeClass('Dropdown-menu--top Dropdown-menu--right');
 
-      $menu.toggleClass('Dropdown-menu--top', $menu.offset().top + $menu.height() > $(window).scrollTop() + $(window).height());
+      $menu.toggleClass('Dropdown-menu--top', top + height > windowSrollTop + windowHeight);
 
-      if ($menu.offset().top < 0) {
+      if (($menu.offset()?.top || 0) < 0) {
         $menu.removeClass('Dropdown-menu--top');
       }
 
-      $menu.toggleClass('Dropdown-menu--right', isRight || $menu.offset().left + $menu.width() > $(window).scrollLeft() + $(window).width());
+      const left = $menu.offset()?.left || 0;
+      const width = $menu.width() || 0;
+      const windowScrollLeft = $(window).scrollLeft() || 0;
+      const windowWidth = $(window).width() || 0;
+
```

Comment:
Null coalesce might be better than a general `||`, in case we decide to change the fallback value from `0` for some reason.